x-pack/elastic-agent/pkg/agent/program/testdata/constraints_config-filebeat.yml

```diff
@@ -8,7 +8,7 @@ filebeat:
     index: logs-generic-default
     processors:
       - add_fields:
-          target: "stream"
+          target: "dataset"
```

Comment:
There a small thing missing here: `dataset.dataset` need to be changed for `dataset.name`, this will create these fields:


1. dataset.type
2. dataset.name
3. dataset.namespace.

@michalpristas @ruflin The field in the agent configuration will also need to changes? Meaning we will need to adjust the index name generation.

Reply:
in configuration i would expect this
```
datasources:
  - namespace: testing
    use_output: default
    inputs:
    - type: docker/metrics
      streams:
        - metricset: status
          dataset: docker.status
```

to become this


```
datasources:
  - namespace: testing
    use_output: default
    inputs:
    - type: docker/metrics
      datasets:
        - metricset: status
          name: docker.status
```

is that correct?
but this feels weird as you can have 

```
datasources:
  - namespace: testing
    use_output: default
    inputs:
    - type: docker/metrics
      datasets:
        - metricset: status
          name: docker.status
        - metricset: cpu
          name: docker.status
```

i dont recall if we agreed to handle configuration change as a separate pack of PRs or the same with this one @ruflin

Reply:
@michalpristas Please handle config changes as a separate PR as it also has effects in Kibana. I plan to open an issue for this. These config changes we should directly in combination with the potential removal of datasources.